src/cheshire/exact.clj

```diff
@@ -0,0 +1,41 @@
+(ns cheshire.exact
+  (:require [cheshire.factory :as factory]
+            [cheshire.parse :as parse]
+            [cheshire.core :as core])
+  (:import (java.io StringReader Reader BufferedReader
+                    Writer)
+           (com.fasterxml.jackson.core JsonFactory)))
+
+(defn- exact-parse [jp parsed]
+  (let [valid-json? (try (nil? (.nextToken jp))
+                         (catch Exception _ false))]
+    (if valid-json?
+      parsed
+      (throw (Exception. "Invalid json")))))
```

Comment:
Can this be `IllegalArgumentException` and the message be `Invalid JSON, expected exactly one parseable object but multiple objects were found"